File: src/lib/slugify.ts
export function slugify(name: string, fallback = "model", maxLen = 40): string {
    let s = name.toLowerCase();
    s = s.replace(/[\s_]+/g, "-");
    s = s.replace(/[^a-z0-9-]/g, "");
    s = s.replace(/-+/g, "-").replace(/^-+|-+$/g, "");
    s = s.slice(0, maxLen).replace(/-+$/, "");
    return s || fallback;
}

File: sidecar/slugify.py
import re


def slugify(name: str, fallback: str = "model", max_len: int = 40) -> str:
    s = name.lower()
    s = re.sub(r'[\s_]+', '-', s)
    s = re.sub(r'[^a-z0-9-]', '', s)
    s = re.sub(r'-+', '-', s).strip('-')
    s = s[:max_len].rstrip('-')
    return s if s else fallback
